src/Analysis/GWASV2/Components/Diagrams/CohortsOverlapDiagram/CohortsOverlapDiagram.test.jsx

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import { screen, render } from '@testing-library/react';
+import '@testing-library/jest-dom/extend-expect';
+import { QueryClient, QueryClientProvider } from 'react-query';
+import CohortsOverlapDiagram from './CohortsOverlapDiagram';
+import ValidState from '../../../TestData/States/ValidState';
+import { SourceContextProvider } from '../../../Utils/Source';
+import {
+  fetchSimpleOverlapInfo,
+  useSourceFetch,
+} from '../../../Utils/cohortMiddlewareApi';
+
+// Mock relevant API calls:
+jest.mock('../../../Utils/cohortMiddlewareApi');
+fetchSimpleOverlapInfo.mockResolvedValue({
+  cohort_overlap: {
+    case_control_overlap: 123,
+  },
+});
+useSourceFetch.mockResolvedValue({
+  sourceId: 2, loading: false,
+});
+
+// Other generic arguments and functions shared by tests below:
+const mockedQueryClient = new QueryClient({
+  defaultOptions: {
+    queries: { retry: false },
+  },
+});
+
+// wrapper render method:
+const renderDiagram = (
+  selectedStudyPopulationCohort, dispatch) => render(
+  <QueryClientProvider client={mockedQueryClient}>
+    <SourceContextProvider>
+      <CohortsOverlapDiagram
+        dispatch={dispatch}
+        selectedStudyPopulationCohort={selectedStudyPopulationCohort}
+        selectedCaseCohort={selectedStudyPopulationCohort}
+        selectedControlCohort={selectedStudyPopulationCohort}
+        selectedCovariates={[]}
+        outcome={null}
+      />
+    </SourceContextProvider>
+  </QueryClientProvider>,
+);
+
+// Tests:
+describe('CohortsOverlapDiagram component', () => {
+  beforeEach(() => {
+    // workaround for issue https://github.com/jsdom/jsdom/issues/1664 / https://stackoverflow.com/questions/53725064/d3node-getcomputedtextlength-is-not-a-function:
```

Comment:
Nice job figuring this out. I don't understand 100% why JSDOM complains, but I pulled your changes locally and confirmed the workaround is needed and after using the work around screen.debug() prints out the DOM containing the SVG once flushPromises has been fulfilled. 

Also read through https://stackoverflow.com/questions/44741102/how-to-make-jest-wait-for-all-asynchronous-code-to-finish-execution-before-expec and it looks like flushPromises = () => new Promise(setImmediate); is a good solution for making Jest wait for all asynchronous code to finish execution before expecting an assertion